src/coreclr/jit/morph.cpp

```diff
@@ -13199,6 +13216,14 @@ GenTree* Compiler::fgMorphSmpOpOptional(GenTreeOp* tree)
 
             if (varTypeIsStruct(typ) && !tree->IsPhiDefn())
             {
+                // Block ops handle assertion kill/gen specially.
+                // See PrepareDst and PropagateAssertions
+                //
+                if (optAssertionPropDone != nullptr)
```

Comment:
So here we say that we are doing `struct1 = struct2` and is candidate for copy prop?

Reply:
No, all that happens over in `optAssertionProp_LclVar` and the new `optAssertionProp_LclFld`. before the tree is morphed.

This check is asking whether or not we've already killed and generated assertions from the post-morph tree.